### ISSUE TITLE
Do not require a port when making a connection

### ISSUE THREAD

**Issue by snotmare** [lbdremy/solr-node-client#327]
We have created a dns entry that forwards to the correct ip:port. We cannot use our dns entry when connecting with this library since it makes a connection assuming you have a port. This is from lib/solr.ts...

```
    this.undiciClient = new UndiciClient(
      `${urlPrefix}${this.options.host}:${this.options.port}`,
      {
        connect: this.options.tls,
      }
    );
```

How best can we change this code to not assume a port? I see you've coded yourself into a corner by defaulting the port. Perhaps it would be best to add another option, "noPort", to handle this?

**Comment by snotmare** [comment]
Hello! Any thoughts on this?

**Comment by kibertoad** [comment]
I will review and finish this over weekend, sorry for the delay.

**Comment by snotmare** [comment]
No worries! I appreciate you taking time for this. Cheers!

**Comment by 58bits** [comment]
I think I may have also just bumped into this. We have an nginx proxy server in front of our Solr installation, and so we don't require a port number, but we do require https (secure: true). We can't seem to get a connection to work if we leave the port off, nor if we set it to 443.

**Comment by kibertoad** [comment]
Implemented slightly less invasive way to support optional port, released in solr-client@0.10.0-rc10

**Comment by SerdarSanri** [comment]
@kibertoad thanks for the update but you forgot the `:` in url. Can we drop the dangling colon if no port is defined or set to null in options
